packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -145,48 +161,125 @@ export const DatePicker: React.VFC<DatePickerProps> = ({
+    }
+  }
+
+  const handleTextChange = e => {
+    const target = e.target as HTMLInputElement
+    setValueString(target.value)
+  }
+
+  const handleKeyDown = e => {
+    // check whether event key is arrow down or alt + arrow down, open calendar if so.
+    if (e.key === "ArrowDown" || (e.key === "ArrowDown" && e.altKey === true)) {
+      e.preventDefault()
+      setIsOpen(true)
+    }
+    return
+  }
+
+  /** Watcher to toggle the current date format when
+   * both values are present.
+   * * */
+  useEffect(() => {
+    if (valueString && valueDate) {
+      const dateNumeral = parse(valueString, DateFormat.numeral, new Date())
+      const dateText = parse(valueString, DateFormat.text, new Date())
+
+      // isValid will return true for "Invalid Date" which is a truthy Date object
+      if (dateNumeral.toString() !== "Invalid Date") {
+        setCurrentDateFormat(DateFormat.numeral)
+      } else if (dateText.toString() !== "Invalid Date") {
+        setCurrentDateFormat(DateFormat.text)
+      }
+    }
+  }, [valueString, valueDate])
+
+  /** On first render, if consumer has passed in a valueDate
+   * we render it as a string within the input.
+   * * */
+  useEffect(() => {
+    if (valueDate && !valueString) {
+      try {
+        // format() can't be passed an invalid date, so we need to catch the range error.
+        const formattedDate = format(new Date(valueDate), DateFormat.text)
+        if (formattedDate.toString() !== "Invalid Date") {
+          setValueString(formattedDate)
+        }
+      } catch (error) {
+        setIsTextValid(false)
+        onChange(undefined)
+      }
+    }
+  }, [])
 
   return (
     <div ref={wrapperRef}>
       <div ref={setReferenceElement}>
-        <Label disabled={isDisabled} htmlFor={id} labelText={labelText} />
-        <button
-          className={cx(
-            datePickerStyles.button,
-            datePickerStyles.withStartIconAdornment,
-            {
-              [datePickerStyles.disabled]: isDisabled,
-            }
-          )}
+        <DateInput
           id={id}
+          inputRef={inputRef}
+          buttonRef={buttonRef}
+          isOpen={isOpen}
+          value={valueString ? valueString : ""}
           disabled={isDisabled}
-          ref={buttonRef}
-          onClick={handleOpenClose}
-          aria-label={
-            value
-              ? `Change date: ${value.toLocaleDateString(
-                  "en-US",
-                  dateFormatOptions
-                )}`
-              : "Choose date"
+          handleOnBlur={() =>
+            valueString !== undefined && handleFormatChange(valueString)
           }
+          onFocus={() =>
+            valueString !== undefined && handleFormatChange(valueString)
+          }
+          labelText={labelText}
+          description="Format: mm/dd/yyyy"
+          icon={dateStart}
+          onButtonClick={handleOpenClose}
+          onChange={handleTextChange}
+          calendarId={"calendar-dialog"}
```

Comment:
This will need to be unique for cases where people put more than one datepicker on a single screen.